azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -477,34 +527,36 @@ def format_message(msg):
                     break
             else:  # since the vmSettings were updated, the response must include an etag
                 message = format_message("The vmSettings response does not include an Etag header")
-                self._vm_settings_error_reporter.report_error(message)
                 raise ProtocolError(message)
 
             response_content = ustr(response.read(), encoding='utf-8')
 
             vm_settings = ExtensionsGoalStateFactory.create_from_vm_settings(response_etag, response_content, correlation_id)
 
             # log the HostGAPlugin version
-            if vm_settings.host_ga_plugin_version != self._host_plugin_version:
-                self._host_plugin_version = vm_settings.host_ga_plugin_version
+            if vm_settings.host_ga_plugin_version != self._version:
+                self._version = vm_settings.host_ga_plugin_version
                 message = "HostGAPlugin version: {0}".format(vm_settings.host_ga_plugin_version)
                 logger.info(message)
                 add_event(op=WALAEventOperation.HostPlugin, message=message, is_success=True)
 
             # Don't support HostGAPlugin versions older than 123
             # TODO: update the minimum version to 1.0.8.123 before release
             if vm_settings.host_ga_plugin_version < FlexibleVersion("1.0.8.117"):
-                raise_not_supported(reset_state=True)
+                raise_not_supported()
 
-            self._host_plugin_supports_vm_settings = True
+            self._supports_vm_settings = True
             self._cached_vm_settings = vm_settings
+
+            if vm_settings.source == GoalStateSource.FastTrack:
```

Comment:
At this point it's all about FastTrack and source always be FastTrack otherwise we would have raised not supported exception. So, I don't see point of having `else `and clearing the state. I don't know if I'm missing something here.

Reply:
The HGAP continues to return both FastTrack and Fabric goal states, nothing has changed there. GoalState.update(), which calls this method is the one that will ignore the Fabric goal states.

Clearing the state is not strictly necessary, since if we receive a Fabric goal state, its timestamp will be greater than the saved state. However, deleting the state keeps things tidy and reduces the code paths the agent goes through during service start (the persisted state is used only during service start if FastTrack is disabled or not supported by the HGAP).

Reply:
Gotcha, Thanks for clarifying.